projects/admin/src/app/guard/can-update.guard.ts

```diff
@@ -0,0 +1,44 @@
+import { CanActivate, ActivatedRouteSnapshot, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { RecordPermission, RecordPermissionService } from '../service/record-permission.service';
+import { ToastrService } from 'ngx-toastr';
+import { TranslateService } from '@ngx-translate/core';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CanUpdateGuard implements CanActivate {
+
+  constructor(
+    private _permissionService: RecordPermissionService,
+    private _toastrService: ToastrService,
+    private _translateService: TranslateService,
+    private _router: Router) {
+  }
+
+  /**
+   * Check if the current logged user can update a resource
+   * @param next - ActivatedRouteSnapshot
+   * @param state - RouterStateSnapshot
+   */
+  canActivate(next: ActivatedRouteSnapshot,
+              state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+
+    this._permissionService.getPermission(next.params.type, next.params.pid).subscribe(
+      (permission: RecordPermission) => {
+        if (!permission.update.can) {
+          this._toastrService.warning(
+            this._translateService.instant('Access denied'),
+            this._translateService.instant(next.params.type)
+          );
+          // Redirect to homepage
+          this._router.navigate(['/']);
```

Comment:
On the jinja side we have a dedicated page for index not found, permission denied, etc. Perhaps for consistency it should be nice to have the same. You can perhaps check with the PO.